cmd/snap-failure/cmd_snapd.go

```diff
@@ -158,14 +181,24 @@ func (c *cmdSnapd) Execute(args []string) error {
 	// before restarting, otherwise the restart command will fail because the
 	// systemd can't create the file
 	// always remove to avoid TOCTOU issues but don't complain about ENOENT
-	err = os.Remove(dirs.SnapdSocket)
-	if err != nil && !os.IsNotExist(err) {
-		logger.Noticef("snapd socket still exists before restarting socket service, but unable to remove: %v", err)
+	for _, fn := range []string{dirs.SnapdSocket, dirs.SnapSocket} {
```

Comment:
from my own testing is a bit unclear if we need this all

Reply:
yes, reading your observations I think it's likely that if these sockets exist at this point, it can only be because the real snapd.service and snapd.socket have them.

For clarity, my original thinking around deleting the socket file was that maybe the failed snapd somehow still has ownership of the socket file(s), but to protect against that I think the smarter thing to do is to delete the socket files _after_ we stop snapd.socket, but _before_ we try to start the ephemeral snapd. Then, with your logic to make sure that the ephemeral snapd properly started the real snapd.service and snapd.socket again, I think we should be fine.

I guess I have some preconceived notion that we want to avoid any kind of waiting in snap-failure for maximum robustness, but I think waiting at the end here after the ephemeral snapd should have cleaned up is fine.

Reply:
We can probably move the remove as you described without impairing too much the behavior vs current shipped snapds.

Reply:
to be clear there's lots that can go wrong with using the current snapds for reverting, but we still need to try to do a best effort.